packages/lesswrong/components/tagging/SubscribeButton.tsx

```diff
@@ -35,26 +37,36 @@ const SubscribeButton = ({
   className?: string,
   classes: ClassesType,
 }) => {
+  const currentUser = useCurrentUser();
+  const { openDialog } = useDialog();
   const { isSubscribed, subscribeUserToTag } = useSubscribeUserToTag(tag)
   const { flash } = useMessages();
   const { captureEvent } = useTracking()
   const { LWTooltip, NotifyMeButton } = Components
-  
+
   const onSubscribe = async (e: React.MouseEvent<HTMLButtonElement>) => {
     try {
       e.preventDefault();
-      const newMode = isSubscribed ? "Default" : "Subscribed"
-      captureEvent('newSubscribeClicked', {tagId: tag._id, newMode})
-      subscribeUserToTag(tag, newMode)
-      
-      flash({messageString: isSubscribed ? "Unsubscribed" : "Subscribed"});
+
+      if (currentUser) {
+        const newMode = isSubscribed ? "Default" : "Subscribed";
+        captureEvent('newSubscribeClicked', {tagId: tag._id, newMode});
+        subscribeUserToTag(tag, newMode);
+
+        flash({messageString: isSubscribed ? "Unsubscribed" : "Subscribed"});
+      } else {
+        openDialog({
+          componentName: "LoginPopup",
+          componentProps: {}
+        });
```

Comment:
I would not have been able to tell you that's how we do it. Mildly surprising API. I hope the componentName is typed.

Reply:
It wasn't. It is now.

Reply:
@jpaddison3 Actually, it might be sensible for you to have one last quick look over this as a sanity check before I merge it. That last commit wasn't complicated, but it did touch a lot of files.